classes/task/queue_to_process_pending_merge_requests.php

```diff
@@ -0,0 +1,72 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+namespace tool_mergeusers\task;
+use \tool_mergeusers\merge_request;
+/**
+ * Version information
+ *
+ * @package     tool
+ * @subpackage  mergeusers
+ * @author      Nicola Vallinoto, Liguria Digitale
+ * @author      Jordi Pujol-Ahulló, SREd, Universitat Rovira i Virgili
+ * @license     http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class queue_to_process_pending_merge_requests extends \core\task\scheduled_task {
+    public function get_name() {
+        return get_string('queue_to_process_pending_merge_requests', 'tool_mergeusers');
+    }
+    /**
+     * Run task for getting merge requests and adding them to adhoc task list.
+     *
+     */
+    public function execute() {
+        mtrace("Task " .$this::class. " started.");
+        global $DB;
+        // Read from moodle table records with status = QUEUED_NOT_PROCESSED.
+        // Add each record to adhoc task.
+        // Update STATUS of each record added to adhoc task.
+        $mergerequestsnotyetscheduled = $DB->get_recordset(merge_request::TABLE_MERGE_REQUEST,
+                                                        ['status' => merge_request::QUEUED_NOT_PROCESSED],
+                                                        $sort = '',
+                                                        $fields = 'id');
```

Comment:
here we could provide an overall overview of how many requests are pending to process, like:

`mtrace('There are ' . count($mergerequestsnotyetscheduled) . ' pending requests to schedule.');`

Reply:
done

Reply:
here it is
Execute scheduled task: Get queue of the pending merge requests and launch adhoc task for each single request (tool_mergeusers\task\queue_to_process_pending_merge_requests)
... started 16:06:53. Current memory use 14.5 MB.
Task tool_mergeusers\task\queue_to_process_pending_merge_requests started.
There are 0 pending requests to schedule.
Task tool_mergeusers\task\queue_to_process_pending_merge_requests completed!
... used 3 dbqueries
... used 44.730632066727 seconds
Scheduled task complete: Get queue of the pending merge requests and launch adhoc task for each single request (tool_mergeusers\task\queue_to_process_pending_merge_requests)
have some problems with the query
      $mergerequestsnotyetscheduled = $DB->get_recordset(merge_request::TABLE_MERGE_REQUEST,
                                                        ['status' => merge_request::QUEUED_NOT_PROCESSED],
                                                        $sort = '',
                                                        $fields = 'id');
no results.

Reply:
From this output, we can say that we have to remove 2 out of 3 mtraces. 

To remove, since this information is already provided by Moodle core:

1. Task tool_mergeusers\task\queue_to_process_pending_merge_requests started.
2. Task tool_mergeusers\task\queue_to_process_pending_merge_requests completed!

To keep:

1. There are 0 pending requests to schedule.

Reply:
What problems did you find with get_recordset?

Reply:
problem solved with get_recordset.